src/common/downloadFileDialog.component.tsx

```diff
@@ -0,0 +1,127 @@
+import WarningIcon from '@mui/icons-material/Warning';
+import {
+  Box,
+  Button,
+  CircularProgress,
+  Dialog,
+  DialogActions,
+  DialogContent,
+  DialogTitle,
+  FormHelperText,
+} from '@mui/material';
+import { CancelledError, useQueryClient } from '@tanstack/react-query';
+import { AxiosError } from 'axios';
+import React from 'react';
+import { APIImage, APIImageWithURL } from '../api/api.types';
+import handleIMS_APIError from '../handleIMS_APIError';
+import { downloadFileByLink } from '../utils';
+
+export interface BaseDownloadFileProps {
+  open: boolean;
+  onClose: () => void;
+  fileType: 'Image' | 'Attachment';
+}
+
+export interface ImageDownloadDialogProps extends BaseDownloadFileProps {
+  fileType: 'Image';
+  getFile: (id: string) => Promise<APIImageWithURL>;
+  file: APIImage;
+}
+
+export type DownloadFileProps = ImageDownloadDialogProps;
+
+const DownloadFileDialog = (props: DownloadFileProps) => {
+  const { open, onClose, fileType, file, getFile } = props;
+
+  const [formError, setFormError] = React.useState<string | undefined>(
+    undefined
+  );
+
+  const [isLoading, setIsLoading] = React.useState(false);
+
+  const queryClient = useQueryClient();
+
+  const handleClose = React.useCallback(() => {
+    queryClient.removeQueries({
+      queryKey: ['Image', file.id],
+    });
+    setFormError(undefined);
+    setIsLoading(false);
+    onClose();
+  }, [onClose, file, queryClient]);
+
+  const handleDownloadFile = React.useCallback(
+    (data: APIImageWithURL | undefined) => {
+      if (data) {
+        downloadFileByLink(document, data.download_url, data.file_name);
+        onClose();
+      } else {
+        setFormError('No data provided. Please refresh and try again');
+      }
+    },
+    [onClose]
+  );
+
+  const handleClick = React.useCallback(async () => {
+    setIsLoading(true);
+
+    let data: APIImageWithURL | undefined = undefined;
+    let closing: boolean = false;
+    try {
```

Comment:
Can you use a `then` and ` catch` block please as we don't tend to use try and catch.